components/builder-jobsrv/src/server/log.rs

```diff
@@ -0,0 +1,44 @@
+use error::Result;
+use std::fs;
+use std::path::PathBuf;
+
+/// Encapsulates the local filesystem directory in which in-process
+/// build job logs will be collected prior to being sent to long-term
+/// storage.
+#[derive(Clone, Debug)]
+pub struct LogDirectory {
+    path: PathBuf,
+}
+
+impl LogDirectory {
+    /// Create a new `LogDirectory` wrapping `path`.
+    pub fn new(path: &str) -> Self {
```

Comment:
A pattern you'll commonly see in any functions which deal with paths is something like this

```
pub fn new<T>(path: T> -> Self where T: AsRef<Path> {
    LogDirectory { path: path.as_ref().to_pathbuf() }
}
```

This will allow you to use various types to interact with the public functions of your API which all can be referenced as a Path